src/odb/src/db/dbBlock.cpp

```diff
@@ -3453,6 +3453,35 @@ void dbBlock::writeGuides(const char* filename) const
   guide_file.close();
 }
 
+void dbBlock::writeMacroPlacement(const std::string& file_name)
```

Comment:
Prefer taking a ostream instead of a string. That way you can unit test this without file IO.

Reply:
I'm going to take that back actually. Since this is getting automatically mapped into SWIG. I think it's potentially better to return a string instead of a write command, and then implement the file writing in TCL.